sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy

```diff
@@ -502,43 +503,51 @@ class APISpec extends Specification {
     to play too nicely with mocked objects and the complex reflection stuff on both ends made it more difficult to work
     with than was worth it. Because this type is just for BlobDownload, we don't need to accept a header type.
      */
-    def getStubResponseForBlobDownload(int code, Flux<ByteBuffer> body, String etag) {
-        return new HttpResponse() {
+    static class MockDownloadHttpResponse extends HttpResponse {
+        private final int statusCode
+        private final HttpHeaders headers
+        private final Flux<ByteBuf> body
+
+        MockDownloadHttpResponse(HttpResponse response, int statusCode, Flux<ByteBuf> body) {
+            this.request(response.request())
+            this.statusCode = statusCode
+            this.headers = response.headers()
+            this.body = body
+        }
 
-            @Override
-            int statusCode() {
-                return code
-            }
+        @Override
+        int statusCode() {
+            return statusCode
+        }
 
-            @Override
-            String headerValue(String s) {
-                return null
-            }
+        @Override
+        String headerValue(String s) {
+            return headers.value(s)
+        }
 
-            @Override
-            HttpHeaders headers() {
-                return new HttpHeaders()
-            }
+        @Override
+        HttpHeaders headers() {
+            return headers
+        }
 
-            @Override
-            Flux<ByteBuffer> body() {
-                return body
-            }
+        @Override
+        Flux<ByteBuf> body() {
+            return body
+        }
 
-            @Override
-            Mono<byte[]> bodyAsByteArray() {
-                return null
-            }
+        @Override
+        Mono<byte[]> bodyAsByteArray() {
+            return Mono.error(new IOException())
+        }
 
-            @Override
-            Mono<String> bodyAsString() {
-                return null
-            }
+        @Override
+        Mono<String> bodyAsString() {
+            return Mono.error(new IOException())
+        }
 
-            @Override
-            Mono<String> bodyAsString(Charset charset) {
-                return null
-            }
+        @Override
+        Mono<String> bodyAsString(Charset charset) {
```

Comment:
Is this necessary to have, it looks like a duplicate to the one above? Since it is a mock response, we can only include the necessary part.

Reply:
You have to include everything as it's implementing an interface, I should change extends -> implements